facilities/validation/gis_helper.rb

```diff
@@ -4,26 +4,35 @@
 
 # Spec helper to fetch GIS data and metadata
 class GISHelper
-  ALL_PARAMS = {
-    where: '1=1',
-    inSR: 4326,
-    outSR: 4326,
-    returnGeometry: true,
-    outFields: '*',
-    f: 'json'
-  }.freeze
 
   METADATA_PARAMS = {
     f: 'json'
   }.freeze
 
-  def self.get_all(url)
+  FACILITY_TYPES = { nca: 'NCA_Facilities',
+                     vba: 'VBA_Facilities',
+                     vha: 'VHA_Facilities',
+                     vc: 'VHA_VetCenters'
+                    }.freeze
+
+  def self.get_all(url,max_record_count)
     conn = Faraday.new(url: url) do |c|
       # c.response :logger
       c.adapter Faraday.default_adapter
     end
-    response = conn.get url, ALL_PARAMS
-    JSON.parse(response.body)
+    data = JSON.parse conn.get(url, params).body
+    count = data['features'].length
+    query_count = 1
+
+    #fetch additional pages when applicable
+    while count == max_record_count
+      response = conn.get(url, params.merge(resultOffset:(query_count * max_record_count)))
+      additional_data = JSON.parse(response.body)
+      data['features'] += additional_data['features']
+      count = additional_data['features'].length
+      query_count += 1
+    end
```

Comment:
I generally try to avoid using for/while/loop in ruby, but an Enumerable alternative doesn't immediately come to mind right now for iterating a paginated collection.

That being said, I wonder if `loop do` with `break if` conditions reads more idiomatically?